### PR TITLE
Fix extension filters & Set specific extension on save

### PR DESCRIPTION
**Pull request by 6543**
Bug to reproduce:
 1. click save doc
 2. enter name without file extension
 3. no extension is added

### CODE REVIEW

**Review by Epsilon02 — APPROVED** [review]

**Review by 6543 — COMMENTED** [review]
:rocket:

File: src-tauri/src/load_save/handler.rs [hunk 1]
```diff
@@ -21,15 +21,31 @@ lazy_static! {
...
+    filename_changed = true;
   }
 
+  if filename_changed {
+    log::debug!("filename changed: {:?}", filename);
+    if filename.exists() {
+      return Err(LoadSaveError::FileExists(filename.to_string_lossy().into()));
+    }
+  }
+  log::debug!("file_name: {:?}", filename.file_name());
```
Comment (crapStone) [inline src-tauri/src/load_save/handler.rs:47]:
```suggestion
```
is already logged in line 42

File: src/app/menubar/menubar.component.ts [hunk 2]
```diff
@@ -76,11 +76,11 @@ export class MenubarComponent implements OnInit {
   loadFile(): void {
     this.tauriService
       .open({
-        filter: '*.cb2', // TODO get filetypes from backend
+        filter: 'cb2', // TODO get filetypes from backend
         multiple: false,
       })
       .subscribe((path) => {
-        this.loadSaveService.loadDocument(path as string).subscribe(
+        this.loadSaveService.loadDocument((path as string).trim()).subscribe(
```
Comment (crapStone) [inline src/app/menubar/menubar.component.ts:83]: if you trim in rust is this necessary?

File: src-tauri/src/load_save/handler.rs [hunk 3]
```diff
@@ -21,15 +21,31 @@ lazy_static! {
 pub fn save_document(file_type: String, filename: PathBuf, document: CaBr2Document) -> Result<()> {
   log::debug!("type: {}", file_type);
   log::debug!("filename: {:?}", filename);
-  log::debug!("doc: {:#?}", document);
+  log::trace!("doc: {:#?}", document);
 
   let mut filename = filename;
```
Comment (crapStone) [inline src-tauri/src/load_save/handler.rs:26]: trim the filename here and not in the frontend.
(never trust your frontend)

File: src/app/menubar/menubar.component.ts [hunk 4]
```diff
@@ -90,10 +90,10 @@ export class MenubarComponent implements OnInit {
   saveFile(): void {
     this.tauriService
       .save({
-        filter: '*.cb2',
+        filter: 'cb2',
       })
       .subscribe((path) => {
-        this.loadSaveService.saveDocument('cb2', path as string, this.modelToDocument()).subscribe(
+        this.loadSaveService.saveDocument('cb2', (path as string).trim(), this.modelToDocument()).subscribe(
```
Comment (crapStone) [inline src/app/menubar/menubar.component.ts:96]: look at previous comment

File: src/app/menubar/menubar.component.ts [hunk 5]
```diff
@@ -103,10 +103,10 @@ export class MenubarComponent implements OnInit {
   exportPDF(): void {
     this.tauriService
       .save({
-        filter: '*.pdf',
+        filter: 'pdf',
       })
       .subscribe((path) => {
-        this.loadSaveService.saveDocument('pdf', path as string, this.modelToDocument()).subscribe(
+        this.loadSaveService.saveDocument('pdf', (path as string).trim(), this.modelToDocument()).subscribe(
```
Comment (crapStone) [inline src/app/menubar/menubar.component.ts:109]: same here

**Review by crapStone — CHANGES_REQUESTED** [review]

File: src-tauri/src/load_save/handler.rs (hunk 3 above)
Comment (6543) [reply src-tauri/src/load_save/handler.rs:26]: doesn't PathBuf already do this?

**Review by crapStone — APPROVED** [review]

**Pull request merged**